### PR TITLE
Pulse weapons are now plasma weapons

### PR DESCRIPTION
**Pull request by MadmanMartian**
As requested. The balancing could do with some tweaking, so feedback is requested

Currently replaces the pulse weapons with plasma weapons, who fire one of three bolts

 - Light: 20 damage
 - Normal: 30 damage
 - Heavy: 40 damage

And has a base armor penetration value of 10

### CODE REVIEW

**Comment by MadmanMartian** [comment]
Requests by Reere:
 - Use the larger icon from tau ceti
 - Use the fade out collision animation that tau ceti's larger projectile uses
 - Have it be coloured by the game, rather than using coloured sprites.
 - Have the colour be fed to the projectile by the firemode

File: code/modules/projectiles/guns/energy/plasma.dm [hunk 1]
```diff
@@ -18,20 +18,20 @@
 	recoil_buildup = 1 //pulse weapons have a bit more recoil
 
 	firemodes = list(
-		list(mode_name="stun", projectile_type=/obj/item/projectile/beam/stun, fire_sound='sound/weapons/Taser.ogg', fire_delay=null, charge_cost=null, icon="stun"),
-		list(mode_name="kill", projectile_type=/obj/item/projectile/beam, fire_sound='sound/weapons/Laser.ogg', fire_delay=null, charge_cost=null, icon="kill"),
-		list(mode_name="DESTROY", projectile_type=/obj/item/projectile/beam/pulse, fire_sound='sound/weapons/pulse.ogg', fire_delay=25, charge_cost=50, icon="destroy"),
+		list(mode_name="maim", projectile_type=/obj/item/projectile/plasma/light, fire_sound='sound/weapons/Taser.ogg', fire_delay=null, charge_cost=null, icon="stun"),
```
Comment (MadmanMartian) [inline code/modules/projectiles/guns/energy/plasma.dm]: Fix this

File: code/modules/projectiles/guns/energy/plasma.dm [hunk 2]
```diff
@@ -40,12 +40,12 @@
...
-		list(mode_name="RAPID", projectile_type=/obj/item/projectile/beam/pulse, fire_sound='sound/weapons/pulse.ogg', fire_delay=10, charge_cost=175, icon="destroy"),
+		list(mode_name="DESTROY", projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=null, charge_cost=null, icon="destroy"),
+		list(mode_name="RAPID", projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=10, charge_cost=175, icon="destroy"),
 	)
 
 
-/obj/item/weapon/gun/energy/pulse/cassad
+/obj/item/weapon/gun/energy/plasma/cassad
 	name = "FS PR \"Cassad\""
 	desc = "\"Frozen Star\" brand energy assault rifle, capable of prolonged combat. When surrender is not an option."
```
Comment (ACCount12) [inline code/modules/projectiles/guns/energy/plasma.dm]: "plasma assault rifle" maybe?

Comment (MadmanMartian) [reply code/modules/projectiles/guns/energy/plasma.dm]: It still technically consumes energy, but sure

**Pull request merged**